crates/holochain/src/core/ribosome/host_fn/must_get_entry.rs

```diff
@@ -0,0 +1,269 @@
+use holochain_wasmer_host::prelude::*;
+use crate::core::ribosome::CallContext;
+use crate::core::ribosome::HostContext;
+use crate::core::ribosome::RibosomeT;
+use holochain_cascade::Cascade;
+use holochain_types::prelude::*;
+use holochain_wasmer_host::prelude::WasmError;
+use std::sync::Arc;
+use holochain_p2p::actor::GetOptions as NetworkGetOptions;
+use holochain_p2p::event::GetRequest;
+
+#[allow(clippy::extra_unused_lifetimes)]
+pub fn must_get_entry<'a>(
+    _ribosome: Arc<impl RibosomeT>,
+    call_context: Arc<CallContext>,
+    input: MustGetEntryInput,
+) -> Result<EntryHashed, WasmError> {
+    let entry_hash = input.into_inner();
+    let network = call_context.host_context.network().clone();
+
+    // timeouts must be handled by the network
+    tokio_helper::block_forever_on(async move {
+        let workspace = call_context.host_context.workspace();
+        let mut cascade = Cascade::from_workspace_network(workspace, network);
+        match cascade
+            .retrieve_entry(entry_hash.clone(),
+            // Set every GetOptions manually here.
+            // Using defaults is dangerous in a must_get as it can undermine determinism.
+            // We want refactors to explicitly consider this.
+            NetworkGetOptions {
+                remote_agent_count: None,
+                timeout_ms: None,
+                as_race: true,
+                race_timeout_ms: None,
+                // Never redirect as the returned entry must always match the hash.
+                follow_redirects: false,
+                // Ignore deletes.
+                all_live_headers_with_metadata: true,
```

Comment:
This should be false as you don't care about metadata right?
```suggestion
                all_live_headers_with_metadata: false,
```